aws/rust-runtime/aws-config/src/sts.rs

```diff
@@ -6,7 +6,23 @@
 //! Credential provider augmentation through the AWS Security Token Service (STS).
 
 mod assume_role;
+
+use crate::connector::expect_connector;
+use crate::provider_config::{HttpSettings, ProviderConfig};
 pub use assume_role::{AssumeRoleProvider, AssumeRoleProviderBuilder};
+use aws_hyper::AwsMiddleware;
+use aws_smithy_client::erase::DynConnector;
+
+impl ProviderConfig {
+    pub(crate) fn sdk_client(&self) -> aws_smithy_client::Client<DynConnector, AwsMiddleware> {
+        aws_smithy_client::Builder::<(), AwsMiddleware>::new()
+            .connector(expect_connector(
+                self.connector(&HttpSettings::default()).clone(),
```

Comment:
Clippy is anti-`Clone` here
```suggestion
                self.connector(&HttpSettings::default()),
```

Reply:
oh yeah. good catch clippy